src/libertem/io/dataset/hdf5.py

```diff
@@ -121,6 +121,11 @@ def get_reader(self):
         )
 
     def _do_initialize(self):
+        if self.ds_path is None:
+            datasets = _get_datasets(self.path)
+            datasets_list = sorted(datasets, key=lambda i: i[1], reverse=True)
+            name, size, shape, dtype = datasets_list[0]
+            self.ds_path = name
```

Comment:
Why was this necessary? `type="auto"` should work without, right?

If this is to make `type="hdf5"` without `ds_path` work, I agree that it is a good idea to have - maybe the duplication with the same code in `detect_params` can be resolved by extracting into a helper function, something like `_guess_ds_path` or similar? I guess for `detect_params`, that helper also needs to return the whole list of datasets etc.

Reply:
Yes precisely, it makes sure `ds_path` can still be omitted when autodetecting, and additionally can guess it when specifying HDF5. I looked at writing a generic function, but, as you write, both places require different info. Plus, one calls the executor, the other already runs on the executor. I figured it would be easier to just duplicate in this case. If you have a better design idea, let me know!

Reply:
Ah, didn't see the executor difference there. Don't have a better idea, I think this is ready to be merged then. The conflicts still need to be resolved though, please let me know if I should go ahead and take care of that